common/models/definition.js

```diff
@@ -128,12 +128,13 @@ module.exports = function(Definition) {
               fileData.name, embedId, config);
           }
 
-          config = Entity.getDataFromConfig(config);
-
           // add extra properties for relations
           config[relation.foreignKey] = fk;
           config[relation.embed.key] = embedId;
```

Comment:
Oh, is this the line where we need the `key` property? I find this rather difficult to spot and reason about. Can you please check if there is a solution that is easier to understand? If we cannot find a better way then we can use what you have here, but I'd like you to at least give this a try.